src/app/ubs/ubs/models/ubs.model.ts

```diff
@@ -175,10 +182,14 @@ export class CAddressData {
     return this.languageService.getLangValue(this.district, this.districtEn);
   }
 
-  setDistrict(place_id: string): void {
+  async setDistrict(place_id: string): Promise<void> {
```

Comment:
Error Handling: Added try-catch blocks to both methods to catch and log errors.
Await in setDistrict: ` async  setDistrict` used in 2 places
1) `onDistrictSelected`  method can be deleted
2) Used await when calling `setDistrict` to ensure it completes before proceeding